Packs/Netcraft_V2/Integrations/Netcraft/Netcraft.py

```diff
@@ -0,0 +1,1036 @@
+            pages += get_page(response)
+            if next_token == stop_on_token:
+                break
+        return pages, response.get(api_token_key)
+
+
+def paginate_with_page_num_and_size(
+    client_func: Callable,
+    *func_args,
+    limit: str | int | None = 50,
+    page: str | int | None = None,
+    page_size: str | int | None = None,
+    pages_key_path: Iterable | None = (),
+    api_limit: int = 1000,
+    api_page_num_key: str = 'page',
+    api_page_size_key: str = 'count',
+) -> list:
+    '''
+    Paginates an API endpoint that accepts "page number" and "page size" parameters
+    using the "limit", "page" and "page_size" args provided by demisto.args() as per the XSOAR pagination protocol.
+
+    Args:
+        client_func (Callable[[...], dict]): The client function that calls the API endpoint.
+        api_params (dict[str, Any]): The parameters to call the endpoint with. The pagination args will be added to this arg.
+        limit (str | int | None): The demisto.args() limit.
+        page (str | None): The demisto.args() page.
+        page_size (str | int | None): The demisto.args() page_size.
+        pages_key_path (Iterable | None): The keys used to extract the "pages" returned by the API.
+                                          i.e. If the API returns a JSON in the format: {'data': {'pages': [1,2,3,4,5]}},
+                                          then pages_key_path=('data', 'pages'). The key path MUST point to a list.
+        api_limit (int, optional): The maximum amount of data returned by the API on a sngle call. Defaults to 1000.
+        api_page_num_key (str, optional): The key used by the API as a page number/index. Defaults to 'page'.
+        api_page_size_key (str, optional): The key used by the API as a page size. Defaults to 'count'.
+
+    Returns:
+        list: The combined pages.
+    '''
+    def get_page(pagination_args: dict) -> list[dict]:
+        response = client_func(*func_args, pagination_args)
+        return dict_safe_get(response, keys=pages_key_path, return_type=list)
+
+    page = arg_to_number(page)
+    page_size = arg_to_number(page_size)
+
+    if page and page_size:
+        pagination_args = {api_page_num_key: page, api_page_size_key: min(api_limit, page_size)}
+        return get_page(pagination_args)
+    else:
+        limit = arg_to_number(limit) or 50
+        pages = []
+        for page in range(1, -(-limit // api_limit) + 1):  # ceiling(limit / api_limit)
+            pagination_args = {api_page_num_key: page, api_page_size_key: api_limit}
+            new_page = get_page(pagination_args)
+            pages += new_page
+            if len(new_page) < api_limit:  # end of the data has been reached
+                break
+        del pages[limit:]  # remove the surplus
+        return pages
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(client: Client) -> str:
+    if PARAMS.get('isFetch') and not arg_to_datetime(PARAMS['first_fetch']):
```

Comment:
what if is left blank? We should support not having this time

Reply:
theres a default